modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/snapshot/SnapshotAwarePartitionDataStorage.java

```diff
@@ -194,21 +195,13 @@ private void handleSnapshotInterference(RowId rowId) {
 
     @Override
     public void close() {
-        cleanupSnapshots();
-    }
-
-    private void cleanupSnapshots() {
-        PartitionSnapshots partitionSnapshots = getPartitionSnapshots();
-
-        partitionSnapshots.acquireReadLock();
-
-        try {
-            partitionSnapshots.ongoingSnapshots().forEach(snapshot -> partitionsSnapshots.finishOutgoingSnapshot(snapshot.id()));
-
-            partitionsSnapshots.removeSnapshots(partitionKey);
-        } finally {
-            partitionSnapshots.releaseReadLock();
+        if (partitionKey instanceof ZonePartitionKey) {
+            // This is a hack for the colocation feature, for zone-based partitions snapshots are cleaned up for a bunch of storages at
```

Comment:
Would it make sense to add a TODO to remove the hack when we completely switch to per-zone partitions?